### PR TITLE
#777 implement reusable select component

### LINKED ISSUE #777: [Internal lib] Research and implement a reusable select component
## Brief description:
We are going to be making an internal library for form elements. Research how you can do that based on the documentation of [shadcn](https://ui.shadcn.com/) and [React-hook-form](https://react-hook-form.com/). The general idea is for these components to be fully reusable and to accept props for everything. So that we can not only use them in the HackAUBG page but in the admin panel in the future. You would also have to write tests with all the edge cases to verify that the dropdown is working correctly. If you have any questions don't hesitate to ask @Aleks1908

### PR DESCRIPTION
**Pull request by asiahaslam**
Created new folder for reusable dropdown component. Inside that folder, I put the dropdown component and a "test form" component to test the dropdown component.

resolves #777

### CODE REVIEW

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/dropdown.tsx:43]: Remove this if it is not used anymore

File: services/web/src/lib/reusable-components/test-form.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/test-form.tsx:1]: This is not a `Next` project. `Use client` should not be here.

File: services/web/src/lib/reusable-components/test-form.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/test-form.tsx:3]: I really like that you made this is a separate component. Do not remove it as of now but before merging this PR we should remove this. Because the PR is only for creating the component itself.

File: services/web/package.json
Comment (Aleks1908) [inline services/web/package.json:15]: These are all downloaded from shadcn. Investigate if all of them are used. If some of them are not follow shadcn documentation on how to remove them.

File: services/web/src/App.tsx
Comment (Aleks1908) [inline services/web/src/App.tsx:13]: Remove this if its not used anymore

**Review by Aleks1908 — CHANGES_REQUESTED** [review]

**Comment by Aleks1908** [comment]
And we are waiting for #779 to be merged so that we can start writing tests

File: services/web/src/lib/reusable-components/test-form.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/test-form.tsx:22]: There is no need to create variables for these strings you can directly put them where you call your component.

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/dropdown.tsx:37]: You have to reserve some space for FormMessage, because currently FormMessage will not be rendered when there are no errors,  but when there is an error it will expand the select which will move the whole page.

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/dropdown.tsx:16]: You will have to make the component receive classnames for everything that can be customised.

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (Aleks1908) [inline services/web/src/lib/reusable-components/dropdown.tsx:24]: When combining predefined classnames with classnames that we will be receiving from props you should use cn. You can import it and see how it works

**Review by Aleks1908 — CHANGES_REQUESTED** [review]

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (asynchroza) [reply services/web/src/lib/reusable-components/dropdown.tsx:4]: Also, I would suggest that you rename it to either DropdownOption or DropdownItem

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (asynchroza) [inline services/web/src/lib/reusable-components/dropdown.tsx:19]: This value should be coming from a component property.

```jsx
export function DropdownComponent({ label, placeholder, className, items, name }: DropdownComponentProps) 
```

File: services/web/src/lib/reusable-components/dropdown.tsx
Comment (asynchroza) [inline services/web/src/lib/reusable-components/dropdown.tsx:29]: The type of item should automatically be inferred. If not, something is wrong with your code.

**Comment by asynchroza** [comment]
Thanks for working on this @asiahaslam. I'm really happy to see people writing some good React code. Thanks to you too @Aleks1908

**Pull request merged**
